projects/core/src/auth/user-auth/services/auth-http-header.service.ts

```diff
@@ -22,8 +42,82 @@ export class AuthHttpHeaderService {
   /**
    * Indicates whether the access token is being refreshed
    */
+  // TODO:#13421 - legacy, remove this flag
   protected refreshInProgress = false;
 
+  /**
+   * Starts the refresh of the access token
+   */
+  protected refreshTokenTrigger$ = new Subject<AuthToken>();
+  /**
+   * Indicates whether the access token is being refreshed
+   */
+  protected refreshInProgress$ = new BehaviorSubject<boolean>(false);
+
+  /**
+   * Indicates whether the logout is being performed
+   */
+  protected logoutInProgress$ = new BehaviorSubject<boolean>(false);
```

Comment:
What if we move this flag to AuthService?

We could set it when we call `coreLogout` method. That might solve the problem with logout guard race conditions. And the logout flag would be correctly set in case of all logout situations (not only because token expires)